asset-transfer-sbe/application-javascript/app.js

```diff
@@ -0,0 +1,354 @@
+
+	// setup the wallet to cache the credentials of the application user, on the app server locally
+	const walletPathOrg1 = path.join(__dirname, 'wallet', 'org1');
+	const walletOrg1 = await buildWallet(Wallets, walletPathOrg1);
+
+	// in a real application this would be done on an administrative flow, and only once
+	// stores admin identity in local wallet, if needed
+	await enrollAdmin(caOrg1Client, walletOrg1, org1);
+	// register & enroll application user with CA, which is used as client identify to make chaincode calls
+	// and stores app user identity in local wallet
+	// In a real application this would be done only when a new user was required to be added
+	// and would be part of an administrative flow
+	await registerAndEnrollUser(caOrg1Client, walletOrg1, org1, Org1UserId, 'org1.department1');
+
+	try {
+		// Create a new gateway for connecting to Org's peer node.
+		const gatewayOrg1 = new Gateway();
+		//connect using Discovery enabled
+		await gatewayOrg1.connect(ccpOrg1,
+			{ wallet: walletOrg1, identity: Org1UserId, discovery: { enabled: true, asLocalhost: true } });
+
+		return gatewayOrg1;
+	} catch (error) {
+		console.error(`Error in connecting to gateway for Org1: ${error}`);
+		process.exit(1);
+	}
+}
+
+async function initGatewayForOrg2() {
+	console.log('\n--> Fabric client user & Gateway init: Using Org2 identity to Org2 Peer');
+	const ccpOrg2 = buildCCPOrg2();
+	const caOrg2Client = buildCAClient(FabricCAServices, ccpOrg2, 'ca.org2.example.com');
+
+	const walletPathOrg2 = path.join(__dirname, 'wallet', 'org2');
+	const walletOrg2 = await buildWallet(Wallets, walletPathOrg2);
+
+	await enrollAdmin(caOrg2Client, walletOrg2, org2);
+	await registerAndEnrollUser(caOrg2Client, walletOrg2, org2, Org2UserId, 'org2.department1');
+
+	try {
+		// Create a new gateway for connecting to Org's peer node.
+		const gatewayOrg2 = new Gateway();
+		await gatewayOrg2.connect(ccpOrg2,
+			{ wallet: walletOrg2, identity: Org2UserId, discovery: { enabled: true, asLocalhost: true } });
+
+		return gatewayOrg2;
+	} catch (error) {
+		console.error(`Error in connecting to gateway for Org2: ${error}`);
+		process.exit(1);
+	}
+}
+
+async function readAsset(assetKey, value, ownerOrg, contractOrg1, contractOrg2) {
+	if (value) {
+		console.log(`\n--> Evaluate Transaction: ReadAsset, - ${assetKey} should have a value of ${value} and owned by ${ownerOrg}`);
+	} else {
+		console.log(`\n--> Evaluate Transaction: ReadAsset, - ${assetKey} should not exist`);
+	}
+	let resultBuffer;
+	resultBuffer = await evaluateTransaction(contractOrg1, assetKey);
+	checkAsset('Org1', assetKey, resultBuffer, value, ownerOrg);
+	resultBuffer = await evaluateTransaction(contractOrg2, assetKey);
+	checkAsset('Org2', assetKey, resultBuffer, value, ownerOrg);
+}
+
+async function evaluateTransaction(contract, assetKey) {
+	try {
+		return await contract.evaluateTransaction('ReadAsset', assetKey);
```

Comment:
Not good to hardcode 'ReadAsset' in the called function... why not just call evaluateTransaction in the parent?

Reply:
done